backend/cmd/headlamp.go

```diff
@@ -761,75 +761,62 @@ func isTokenAboutToExpire(token string) bool {
 	return time.Until(expiryTime) <= JWTExpirationTTL
 }
 
-//nolint:funlen
-func refreshAndCacheNewToken(oidcAuthConfig *kubeconfig.OidcConfig,
-	cache cache.Cache[interface{}], token string,
-) (string, error) {
-	const ExtendRefreshTokenTTL = 10 // seconds
-
-	// get provider
-	provider, err := oidc.NewProvider(context.Background(), oidcAuthConfig.IdpIssuerURL)
+func refreshAndCacheNewToken(clientID string, clientSecret string, refreshToken string, issuerURL string) (*oauth2.Token, error) {
+	// Get OIDC provider token URL
+	provider, err := oidc.NewProvider(context.Background(), issuerURL)
 	if err != nil {
-		logger.Log(logger.LevelError, nil, err, "failed to get provider")
-
-		return "", err
+		return nil, fmt.Errorf("error getting provider: %v", err)
 	}
 
-	// get refresh token from cache
-	refreshToken, err := cache.Get(context.Background(), fmt.Sprintf("oidc-token-%s", token))
-	if err != nil || refreshToken == "" {
-		logger.Log(logger.LevelError, nil, err, "failed to get refresh token")
-
-		return "", err
+	token, err := getNewTokenFromRefresh(clientID, clientSecret, refreshToken, provider.Endpoint().TokenURL)
```

Comment:
name of variable should ne clear since this is a refreshToken. So please change it back